metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -506,6 +511,25 @@ def __init__(
                 x.strip()
                 for x in (self.config.ignore_upstream_lineage_platforms.split(","))
             ]
+        else:
+            # return empty list if the config is not set
+            self.ignore_upstream_lineage_platforms = []
+
+        # if ignore upstream lineage platforms doesn't contain "postgres", then consult the upstream_postgres_database_whitelist
+        if "postgres" not in self.ignore_upstream_lineage_platforms:
+            if self.config.upstream_postgres_database_whitelist:
+                self.upstream_postgres_database_whitelist = [
+                    x.strip()
+                    for x in (
+                        self.config.upstream_postgres_database_whitelist.split(",")
+                    )
+                ]
+            else:
+                # return empty list if the config is not set
+                self.upstream_postgres_database_whitelist = []
+        else:
+            self.upstream_postgres_database_whitelist = []
```

Comment:
Could simplify the if/else here.
```suggestion
        if (
            "postgres" not in self.ignore_upstream_lineage_platforms 
            and self.config.upstream_postgres_database_whitelist
        ):
            self.upstream_postgres_database_whitelist = [
                x.strip()
                for x in (
                    self.config.upstream_postgres_database_whitelist.split(",")
                )
            ]
        else:
            # return empty list if the config is not set
            self.upstream_postgres_database_whitelist = []
```